### DISCUSSION TITLE
Enhance Filter in Tabular Layout to fully support API Filter Rules

### DISCUSSION THREAD

**Discussion by phampphi** [directus/directus#6464] [Feature Requests]
Filters can be passed through Tabular Layout to query the items with specific conditions.
However, currently the filters are only supported with logical `AND` i.e the conditions are joined with each other using logical `AND`.
It is not possible to perform query with logical `OR` (`field1=value1 OR field2=value2`).
It is also not possible to perform query with group like `(field1=value2 AND field2=value2) OR (field3=value3 AND field4=value4)`

With this limitation, the use of Tabular layout is restricted. It would be much better if the Tabular Layout could fully support API Filter Rules.

The reason behinds this limitation is the Tabular Layout is current accepts an array of Filter object as input prop.
`export type Filter = {
	key: string;
	field: string;
	operator: FilterOperator;
	value: string;
	locked?: boolean;
}`
This array of Filter object is then parsed by the `filters-to-query` utility function which builds an API query object.
However, this `filters-to-query` utility function simply joins the filters together with logical `AND`.

There are a few ideas to improve this limitation
1. Instead of an array of Filter object, the Tabular layout could accept a single object that conform with the [API Filter Rules spec](https://docs.directus.io/reference/filter-rules/). Then the job of the Tabular layout would be to pass this filter object into the API call to retrieve the items.
2. An less-impact approach would be to modify the  `filters-to-query` utility to take into account an special convention:
`{"key":"api-query-object", "operation":"eq", value: {"first_name": {"_eq": "Rijk"}}}`
The idea is handle an special-convention Filter object which hold a API query object as value. This value can then be directly passed into the API call

This enhancement would benefit the following usecase:
- use of Tabular layout in custom interface
- open to fully support of API Filter Rules in Advanced Filter

Please let me know your thoughts. Thanks

**Comment by rijkvanzanten** [comment]
Yup! 100% agree. We should replace that app internal filter structure to match the api filter rules

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
